controllers/clustergroupupgrade_controller.go

```diff
@@ -991,6 +992,7 @@ func (r *ClusterGroupUpgradeReconciler) doManagedPoliciesExist(
 
 				if len(clustersNonCompliantWithPolicy) == 0 {
 					managedPoliciesCompliantBeforeUpgrade = append(managedPoliciesCompliantBeforeUpgrade, foundPolicy.GetName())
+					managedPoliciesInfo.compliantPolicies = append(managedPoliciesInfo.compliantPolicies, foundPolicy)
```

Comment:
I wonder this would benefit from moving into a more general place. This block is only called if `filterNonCompliantPolicies` is set to true. i.e a dedicated function which is always called to just grabs all policies needed for precaching operator?

Also  can `managedPoliciesCompliantBeforeUpgrade` and `managedPoliciesInfo.compliantPolicies` be combined into into one? One with just the names and another with the full objects? I think it better to merge them to reduce the complexity.

Reply:
One is a string array as part of CGU status. So it's part of the API now. The other one is the real policy object that's local to this reconcile.

Reply:
Gotcha. 

So my concern here is that you're populating `managedPoliciesInfo.compliantPolicies` in a func `doManagedPoliciesExist`. The name implies that it's a general func that can be called multiple times by whoever needs to verify that if policies exists. And if that's the case `managedPoliciesInfo.compliantPolicies` maybe populated as many times as well. Which is why I think it's better move this line and populate it in a more explicit way.

Reply:
Not following. 'managedPoliciesInfo' is supposed to be recreated in each call and returned.

Reply:
oh then all good with this.